### DISCUSSION TITLE
What should I add next?

### DISCUSSION THREAD

**Discussion by notAperson535** [notAperson535/OneClick-macOS-Simple-KVM#3] [Q&A] [answered]
What do you think I should add next to improve/enhance this repository?

**Comment by danryu** [comment, marked as answer]
Nice repo, very useful utility for getting set up fast with a macOS guest VM for use with Xcode etc.

Personally I would like:
- advice on how to resize a disk volume (I use qemu-img resize but the changes aren't reflected after macOS reboot)
- maybe more tips on how to integrate with virt-manager (ie transferring the config from the handy basic.sh script)

**Reply by notAperson535** [reply]
@danryu 

> * advice on how to resize a disk volume (I use qemu-img resize but the changes aren't reflected after macOS reboot)

I am currently working on this right now. Will notify you when it is ready.

> * maybe more tips on how to integrate with virt-manager (ie transferring the config from the handy basic.sh script)

This should already happen with `make.sh`

**Reply by danryu** [reply]
I did use make.sh and it mostly seems to have worked fine - I just don't know if I need to do anything extra now to enable vfio-pci for graphics accel and sound etc

**Reply by notAperson535** [reply]
> I did use make.sh and it mostly seems to have worked fine - I just don't know if I need to do anything extra now to enable vfio-pci for graphics accel and sound etc

for graphics acceleration use gpu passthrough, I don't know about vfio-pci. Sound should work but not great. You could get a cheap USB sound card and pass that through.

**Reply by notAperson535** [reply]
forgot to ping @danryu

**Reply by danryu** [reply]
Yes I am using a USB external soundcard as a workaround, works fine.
Not sure about the graphics accel though - in your passthrough doc at https://oneclick-macos-simple-kvm.notaperson535.is-a.dev/docs/passthrough/ you specify using vfio-pci kernel module, modifying GRUB kernel boot parameters, and attaching the card to QEMU via cmdline arguments.
I'm not sure how relevant all of this is, with reference to current virt-manager (see this comment here: https://www.reddit.com/r/Fedora/comments/jts8m7/comment/gc7v4b8/?utm_source=share&utm_medium=web2x&context=3).
I don't want to start changing kernel driver configuration and boot parameters unless absolutely necessary...

**Reply by notAperson535** [reply]
@danryu follow a guide for passing thorugh gpu to virt-manager, don't follow the one I have, you need a more specific one.

**Comment by notAperson535** [comment]
@danryu done! [here is the link to the resizing guide](https://oneclick-macos-simple-kvm.notaperson535.is-a.dev/docs/resizing/)

**Reply by danryu** [reply]
Thanks this works great

**Comment by danryu** [comment]
One other thing :)

It would be good to get some guidance on clipboard copy-paste (ie Linux host<->MacOS guest). Not sure if it requires Spice client in macOS or not.

**Reply by notAperson535** [reply]
@danryu remember that macOS uses command+c instead of ctrl+c (command is the start key on your keyboard), look here: https://wiki.qemu.org/Features/Copy-Paste_Integration

**Reply by notAperson535** [reply]
and this: https://networksec.info/gns3/how-to-enable-bidirectional-copy-and-paste-between-the-qemu-vm-and-the-linux-host-running-gns3/2020/

**Reply by hockeymikey** [reply]
Link 404s. I use kdeconnect personally.

**Reply by notAperson535** [reply]
> Link 404s. I use kdeconnect personally.

I do not understand what you mean by that.

**Reply by hockeymikey** [reply]
The link you posted is 404ed. KDEConnect: https://kdeconnect.kde.org/ I use it on all of my devices.

**Reply by notAperson535** [reply]
> The link you posted is 404ed. KDEConnect: https://kdeconnect.kde.org/ I use it on all of my devices.

Oh, I understand now.

**Comment by ichlubna** [comment]
Thanks for this!!! This is one of the most amazing things I found recently. I've been using VirtualBox with MacOS to build and distribute our non-profit mobile APP that helps people with mental health issues. A few days ago, my virtual machine broke and I couldn't make it work, not even after reinstall. This repo saved me! So easy to use!
I don't need anything new but the most crucial thing for me is to keep the OS updated to support the newest Xcode. So basically the continuous support for new versions of Mac. The current Monterey is working great! Thanks again!
By the way, upgrading to new versions is possible only via the setup scripts or can it be done directly in the virtual system by downloading and installing the update?

**Reply by notAperson535** [reply]
> By the way, upgrading to new versions is possible only via the setup scripts or can it be done directly in the virtual system by downloading and installing the update?

As long as you download the latest version of the repo like [this](https://oneclick-macos-simple-kvm.notaperson535.is-a.dev/docs/start-here#important), you should be able to update macOS from inside of the VM.

**Comment by ErichPetersen** [comment]
Congratulations on the project, it has helped me a lot.
One thing I miss is the possibility of using 2 monitors.

**Comment by therealloudking** [comment]
My aim is Sonoma I seen the site icon and I feel that it is possible. My m1 went down yesterday. And was talking with Apple support about sending it in for a battery replacement. But during the rebooting it shut off and had the old ! , i really did like the flexiablity arm and rosetta 2 gave to emulate pretty much run anything i wanted. I have the have the previous usb installer and efi bootable external usb and time machine backups. Got a hp 255 g8 notebook pc that i am quite familiar with. Windows Server 2022 wsl Ubuntu Nobel and Mac Sonoma i believe i can fit on a 250gb. Maybe try my hand at a dual boot with open core i still have my secondary ssd available. ty btw i was just reading and came this way some how. And just maybe show apple support a little of what i know.
